python/ray/data/_internal/execution/interfaces/execution_options.py

```diff
@@ -99,42 +106,74 @@ def object_store_memory(self) -> float:
     def object_store_memory(self, value: float):
         self._object_store_memory = value
 
+    @property
+    def memory(self) -> float:
+        if self._memory is not None:
+            return self._memory
+        return 0.0 if not self._default_to_inf else float("inf")
```

Comment:
Wait why do we want to fallback to anything in here?

Reply:
Often, you won't specify this (or other) resource value. For example, `ExecutionResources(cpu=1)`. In this case, memory can be `float('inf')` or 0 depending on whether you used `ExecutionResources.for_limits(...)`